### ISSUE TITLE
How to distinguish patched from un-patched fields

### ISSUE THREAD

**Issue by datagitlies** [crnk-project/crnk-framework#418]
From [the documentation](http://www.crnk.io/releases/stable/documentation/#_resourcerepositoryv2):

> ### 5.1. ResourceRepositoryV2
> `save(S resource)` Saves a resource upon a `PATCH` request. The general semantics is identical to the `create(…​)` method, with two notable exceptions. First, resources are updated but are not allowed to be inserted. A `ResourceNotFoundException` must be thrown if the resource does not exist. Second, the `PATCH` request allows for partial updates. Internally Crnk will get the current state of a resource, patch it and pass it to this `save` method. **API to distinguish patched from un-patched fields is not yet (directly) available, feel free to open a ticket.**

It sounds like this capability is not "directly" available... what does that mean exactly? Can anyone provide an example?

**Comment by remmeier** [comment]
needs to be clarified, any maybe supported. But what you can do is setup a ResourceModificationFilter and collect all changes. For example in a Thread Local and then query it wherever you need.

**Comment by datagitlies** [comment]
@remmeier thanks for clarifying. I'm wanting an instance of the existing resource (with no changes applied) without having to retrieve it (again) inside of the `save` method of the resource repository. Thus, I will be able to "compare" the existing resource and the patched resource side by side (a very handy feature if you ask me).

So if I understand the `ResourceModificationFilter` correctly, the `resource` instance that is passed into the `modifyAttribute` method is subsequently changed. Meaning, only the first invocation will provide the existing (unchanged) resource, correct? This means I cannot simply hold onto the reference in a Thread Local... I'd have to create a (deep) copy of it first. Is that right?

**Comment by datagitlies** [comment]
Also, would a `ResourceModificationFilter` be picked up automatically via `SpringServiceDiscovery` or do I need to add it to a new `SimpleModule` bean?

**Comment by remmeier** [comment]
yes, ResourceModificationFilter is continuously invoked while the "current" resources is updated with the new patched fields. So the resource gets changed. You can hold on to the set of fields that were changed to do a comparison. 

Both means of registration work fine.
